File: scratch_check_excel.mjs
import * as xlsx from 'xlsx';

const files = [
    "F:\\2.ML_Project\\Demo\\4.tanggal_berjalan\\Historis_24_Maret\\3_Skenario\\Master_report_scenario\\historis_rank_alignment_report.xlsx",
    "F:\\2.ML_Project\\Demo\\4.tanggal_berjalan\\Historis_24_Maret\\3_Skenario\\Arsip\\Pendukung\\historis_rank_alignment_report.xlsx"
];

for (const file of files) {
    console.log(`\n--- Checking ${file} ---`);
    try {
        const workbook = xlsx.readFile(file);
        console.log(`Sheets: ${workbook.SheetNames.join(', ')}`);
        
        for (const sheetName of workbook.SheetNames) {
            const worksheet = workbook.Sheets[sheetName];
            const json = xlsx.utils.sheet_to_json(worksheet, { defval: null });
            
            if (json.length === 0) continue;
            
            const cols = Object.keys(json[0]).filter(c => 
                c.toLowerCase().includes('planned') || 
                c.toLowerCase().includes('2026') || 
                c.toLowerCase().includes('target')
            );
            console.log(`  Sheet '${sheetName}' relevant columns: ${cols.join(', ')}`);
            console.log(`  Sheet '${sheetName}' - Total Rows: ${json.length}`);
            
            let planned_any = 0;
            let planned_tender = 0;
            let planned_pl = 0;
            
            for (const row of json) {
                if (row['planned_any_2026'] === 1 || row['planned_any_2026'] === '1' || row['planned_any_2026'] === true || String(row['planned_any_2026']).toLowerCase() === 'true') {
                    planned_any++;
                }
                if (row['planned_tender_2026'] === 1 || row['planned_tender_2026'] === '1' || row['planned_tender_2026'] === true || String(row['planned_tender_2026']).toLowerCase() === 'true') {
                    planned_tender++;
                }
                if (row['planned_pl_2026'] === 1 || row['planned_pl_2026'] === '1' || row['planned_pl_2026'] === true || String(row['planned_pl_2026']).toLowerCase() === 'true') {
                    planned_pl++;
                }
            }
            
            if (cols.includes('planned_any_2026')) console.log(`    planned_any_2026 positives: ${planned_any}`);
            if (cols.includes('planned_tender_2026')) console.log(`    planned_tender_2026 positives: ${planned_tender}`);
            if (cols.includes('planned_pl_2026')) console.log(`    planned_pl_2026 positives: ${planned_pl}`);
        }
    } catch (e) {
        console.error(`Error reading file:`, e.message);
    }
}

File: scratch_check_excel.py
import pandas as pd
import sys

files = [
    r"F:\2.ML_Project\Demo\4.tanggal_berjalan\Historis_24_Maret\3_Skenario\Master_report_scenario\historis_rank_alignment_report.xlsx",
    r"F:\2.ML_Project\Demo\4.tanggal_berjalan\Historis_24_Maret\3_Skenario\Arsip\Pendukung\historis_rank_alignment_report.xlsx"
]

for file in files:
    print(f"--- Checking {file} ---")
    try:
        xl = pd.ExcelFile(file)
        print(f"Sheets: {xl.sheet_names}")
        for sheet in xl.sheet_names:
            df = xl.parse(sheet)
            cols = [c for c in df.columns if 'planned' in str(c).lower() or '2026' in str(c).lower() or 'target' in str(c).lower()]
            print(f"  Sheet '{sheet}': relevant cols -> {cols}")
            
            # check the total rows, total planned_any_2026 positives, etc if the column exists
            print(f"  Sheet '{sheet}' - Total Rows: {len(df)}")
            if 'planned_any_2026' in df.columns:
                print(f"    planned_any_2026 positives: {df['planned_any_2026'].sum()}")
            if 'planned_tender_2026' in df.columns:
                print(f"    planned_tender_2026 positives: {df['planned_tender_2026'].sum()}")
            if 'planned_pl_2026' in df.columns:
                print(f"    planned_pl_2026 positives: {df['planned_pl_2026'].sum()}")
                
    except Exception as e:
        print(f"Error reading file: {e}")
